src/librustdoc/clean/mod.rs

```diff
@@ -1109,7 +1109,7 @@ fn clean_function<'tcx>(
         };
         let mut decl = clean_fn_decl_with_args(cx, sig.decl, args);
         if sig.header.is_async() {
-            decl.output = decl.sugared_async_return_type();
+            decl.output = decl.output.sugared_async_return_type();
```

Comment:
Since it's now done in `clean_fn_decl_with_args`, is it still needed?

Reply:
It's not done in `clean_fn_decl_with_args` (HIR) but in `clean_fn_decl_from_did_and_sig` (metadata), so it's still necessary when cleaning the HIR.

Reply:
Oh I see. Why is it not done in `clean_fn_decl_with_args`?

Reply:
Good idea, done!